include/nuttx/notifier.h

```diff
@@ -0,0 +1,197 @@
+/****************************************************************************
+ * include/nuttx/notifier.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_NUTTX_NOTIFIER_H
+#define __INCLUDE_NUTTX_NOTIFIER_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/irq.h>
+
+#include <debug.h>
+#include <errno.h>
+
+#include <sys/types.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define ATOMIC_NOTIFIER_INIT(name) {.head = NULL }
+
+#define ATOMIC_NOTIFIER_HEAD(name) \
+  struct atomic_notifier_head name = ATOMIC_NOTIFIER_INIT(name)
+
+/****************************************************************************
+ * Public Type Definitions
+ ****************************************************************************/
+
+struct notifier_block;
+
+typedef int (*notifier_fn_t)(FAR struct notifier_block *nb,
+                             unsigned long action, FAR void *data);
+
+struct notifier_block
+{
+  notifier_fn_t              notifier_call;
+  FAR struct notifier_block *next;
+  int                        priority;
+};
+
+struct atomic_notifier_head
+{
+  FAR struct notifier_block *head;
+};
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+#ifndef __ASSEMBLY__
+#undef EXTERN
+#if defined(__cplusplus)
+#define EXTERN extern "C"
+extern "C"
+{
+#else
+#define EXTERN extern
+#endif
+
+#define notifier_chain_register(nhead, node, unique_priority) \
+  do \
+    { \
+      FAR struct notifier_block **nl = nhead; \
```

Comment:
```suggestion
      FAR struct notifier_block **nl = (nhead); \
```
add  () for all macro parameter